dbt/include/teradata/macros/materializations/incremental/incremental.sql

```diff
@@ -43,6 +43,11 @@
 
 {{ run_hooks(post_hooks, inside_transaction=True) }}
 
+-- implemented this for the grants feature
+{%- set grant_config = config.get('grants') -%}
+{% set should_revoke = should_revoke(existing_relation, full_refresh_mode) %}
```

Comment:
How full_refresh_mode value is coming?

Reply:
DBT core calls **should_full_refresh()** and assigns the return value to full_refresh_mode.
{%- set full_refresh_mode = (should_full_refresh()  or existing_relation.is_view) -%}